### PR TITLE
Updated Mirrormaker2 docs #3204

### PR DESCRIPTION
**Pull request by Koushikk-24**
Signed-off-by: Rupesh <punnarupeshkoushik@gmail.com>

### Updated docs don't describe how to enable compression in the target cluster

_Select the type of your PR_
- Documentation

### Description

This PR updated the Mirror maker 2 docs [clusters and mirrors] issue #3204 

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Review by scholzj — CHANGES_REQUESTED** [review]
Thanks for the PR. But I do not think it really solves the issue since it does not change docs, only example files. In addition:
* You should not change anything in the `examples` directory - these are released files as noted there. If you want to change an example, please do it in `packaging/examples`.
* You took an example of how to use custom replication policy and completely removed the custom replication policy. This example is there for a reason. You cannot just remove something to add something completely different. If you want to add an example showing how to add compression, fine. But add a new file for it and name it accordingly.
* All the example follow the same style how things are named etc. Do not change it. Use the same aliases and basic options as all other MM2 examples.

If you want to add this to the actual docs, you can find them here: https://github.com/strimzi/strimzi-kafka-operator/tree/main/documentation ... there is also a guide about how to contribute to the docs: https://strimzi.io/contributing/guide/

**Comment by Koushikk-24** [comment]
@scholzj, thanks for the review. I have updated the doc file

**Comment by scholzj** [comment]
You still need to revert the changes tot he example files. It would also need some explanation - you should not enable the comperession like this without explaining it as it might be confusing and cause problems to users. And the same as what I said for the example applies also for the docs ... do not change everything what is completely unrelated to the issue.

Maybe @PaulRMellor or @laidan6000 can point you to the right place where to add it.

**Comment by Koushikk-24** [comment]
@scholzj, I have reverted the changes of example files and updated the doc file. Can you please review once again?

File: documentation/modules/configuring/proc-config-mirrormaker2-replication.adoc [hunk 1]
```diff
@@ -34,14 +34,26 @@ spec:
...
-    sourceConnector: {}
+  - sourceCluster: "my-source-cluster"
+    targetCluster: "my-target-cluster"
+    sourceConnector:
+      config:
+        replication.factor: 1
+        offset-syncs.topic.replication.factor: 1
+        sync.topic.acls.enabled: "false"
+        replication.policy.separator: ""
+        replication.policy.class: "io.strimzi.kafka.connect.mirror.IdentityReplicationPolicy
```
Comment (scholzj) [inline documentation/modules/configuring/proc-config-mirrormaker2-replication.adoc:56]: Why are we adding this configuration here? It seems to be unrelated.

File: documentation/modules/configuring/proc-config-mirrormaker2-replication.adoc [hunk 2]
```diff
@@ -34,14 +34,26 @@ spec:
...
-  - alias: "my-cluster-source"
-    bootstrapServers: my-cluster-source-kafka-bootstrap:9092
-  - alias: "my-cluster-target"
-    bootstrapServers: my-cluster-target-kafka-bootstrap:9092
+  - alias: "sourceCluster"
+    bootstrapServers: my-source-cluster-bootstrap:9094
+    config:
+      config.storage.replication.factor: 1
+      offset.storage.replication.factor: 1
+      status.storage.replication.factor: 1
```
Comment (scholzj) [inline documentation/modules/configuring/proc-config-mirrormaker2-replication.adoc:42]: Why are we adding this configuration here? It seems to be unrelated.

File: documentation/modules/configuring/proc-config-mirrormaker2-replication.adoc [hunk 3]
```diff
@@ -34,14 +34,26 @@ spec:
   version: {DefaultKafkaVersion}
   connectCluster: "my-cluster-target"
   clusters:
-  - alias: "my-cluster-source"
-    bootstrapServers: my-cluster-source-kafka-bootstrap:9092
-  - alias: "my-cluster-target"
-    bootstrapServers: my-cluster-target-kafka-bootstrap:9092
+  - alias: "sourceCluster"
```
Comment (scholzj) [inline documentation/modules/configuring/proc-config-mirrormaker2-replication.adoc:37]: Please use the original name. This now anyway breaks the whole example because you change the aliases here but the references use the old names.

File: documentation/modules/configuring/proc-config-mirrormaker2-replication.adoc [hunk 4]
```diff
@@ -34,14 +34,26 @@ spec:
...
+    bootstrapServers: my-source-cluster-bootstrap:9094
+    config:
+      config.storage.replication.factor: 1
+      offset.storage.replication.factor: 1
+      status.storage.replication.factor: 1
+  - alias: "targetCluster"
```
Comment (scholzj) [inline documentation/modules/configuring/proc-config-mirrormaker2-replication.adoc:43]: Please use the original name.

File: documentation/modules/configuring/proc-config-mirrormaker2-replication.adoc [hunk 5]
```diff
@@ -34,14 +34,26 @@ spec:
...
+    bootstrapServers: my-target-cluster-bootstrap:9092
+    config:
+      producer.compression.type: zstd
```
Comment (scholzj) [inline documentation/modules/configuring/proc-config-mirrormaker2-replication.adoc:46]: As I said before. The easiest way to add this would be to add a small paragraphs saying something like _If you want to use compression to ...._ And add a separate snippet with this to it.

**Review by scholzj — CHANGES_REQUESTED** [review]

**Comment by Koushikk-24** [comment]
> @scholzj, I have reverted the changes of example files and updated the doc file. Can you please review once again?

Thanks for the review,  it looks like I still didn't properly understand the solution for this issue.
